variable/include/scipp/variable/variable.h

```diff
@@ -139,6 +141,9 @@ class SCIPP_VARIABLE_EXPORT Variable {
 
   [[nodiscard]] Variable as_const() const;
 
+  auto &unchecked_dims() { return m_dims; }
+  auto &unchecked_strides() { return m_strides; }
```

Comment:
I don't like this. Does anyone have a nicer idea, apart from making all functions which need to access this members? @jl-wynen

Reply:
Are they really needed? The check is not particularly expensive.

Reply:
Which check? The point is that we have no way of setting strides or dims right now, without (in this case) unintended side effects (such as resizing the buffer).

Reply:
Basically, there is a different check for every shape-related operation (such as broadcast, fold, flatten, reshape).

Reply:
Ah, I though you wanted to circumvent the `!*this` check in `dims()`.

The best I can think of right now is a setter with a scary sounding name (to discourage usage). Might be a bit nicer than the getters. But not by much...